### PR TITLE
Remove use of RegisterClass

### PR DESCRIPTION
**Pull request by deltakosh**
Ok folks! another new monster :)

I removed the need for side-effects for a LOT OF classes by replacing RegisterClass with the use of require.context

@ryantrem: please make sure that it is not a problem for the Viewer
@RaananW: I did not do the change for the FlowGraph because for some reasons that I do not know you are using FGxxx in the key for RegisterClass

The main class:
https://github.com/BabylonJS/Babylon.js/pull/15671/files#diff-6a2ff5432bcbb1632f27a94f714a7a50f7b3c4cac5d35bd7b7f8570746752e12

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Meshes/Node/Blocks/geometryCollectionBlock.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { NodeGeometryBlock } from "../nodeGeometryBlock";
 import type { NodeGeometryConnectionPoint } from "../nodeGeometryBlockConnectionPoint";
-import { RegisterClass } from "../../../Misc/typeStore";
+import {} from "../../../Misc/typeStore";
```
Comment (sebavan) [inline packages/dev/core/src/Meshes/Node/Blocks/geometryCollectionBlock.ts:3]: should be removed I guess :-)

File: packages/dev/core/src/contextHelper.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+/**
+ * Extract a class from the bundle context
+ * @param name name of the class to extract
+ * @returns the class or undefined if not found
+ */
+export function ExtractClass(name: string) {
+    if (!_DiscoveredTypes) {
+        _DiscoveredTypes = {};
+        // Use require.context to import all modules
+        const requireModule = require.context(
```
Comment (sebavan) [inline packages/dev/core/src/contextHelper.ts:54]: Is this not webpack specific ? cause this would basically fail for anybody bundling without Webpack ?

Comment (deltakosh) [reply packages/dev/core/src/contextHelper.ts:54]: This is THE question:) hence why I wanted @ryantrem to check

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/contextHelper.ts (hunk 2 above)
Comment (ryantrem) [reply packages/dev/core/src/contextHelper.ts:54]: I can pull your branch and test it out with the viewer (which uses rollup) to be sure.

**Comment by RaananW** [comment]
require.context is not ecma standard, and is webpack specific. adding this will make our package not work in browser anymore. Let's find a better way instead of that.

The idea of require.context is to be a synchronous import, but this is bundler specific (webpack in this case).

**Comment by RaananW** [comment]
> @RaananW: I did not do the change for the FlowGraph because for some reasons that I do not know you are using FGxxx in the key for RegisterClass

This will be changed as part of the changes I am making. there was a thought to shorten the names of the classes to save a few bytes, but this is not the right place to minimize input :-)

File: packages/dev/core/src/contextHelper.ts (hunk 2 above)
Comment (RaananW) [reply packages/dev/core/src/contextHelper.ts:54]: I commented about it. We can't use require.context if we want to stay esm friendly

**Comment by deltakosh** [comment]
Hey this is basically my question. Rollup and parcel have no equivalent so if we want to be side effects free what are our options regarding RegisterClass within the framework?

**Comment by ryantrem** [comment]
typeStore (e.g. `RegisterClass`, `GetClass`) is not considered a public API, it is internal only, correct? If the goal is to be able to tree shake out unused stuff, then it seems like we just need to replace this mechanism with well defined imports, right? I don't know the history though of typeStore and why we even have this functionality to register and retrieve types by names...

**Comment by deltakosh** [comment]
It is internal to allow us to parse from serialized data (as the class will store its type so we can rebuild it at parse time). The question is then: how do you replace the feature?

**Pull request closed without merging**